### PR TITLE
Introduce v1beta1 api version

### LINKED ISSUE #231: Implement v1beta1 API version
Lets create a new API version (**v1beta1**) so:

- We can make some API changes for things like ClusterClass
- We guarantee the stability of the API for our consumers.

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:
This change adds v1beta1 api version and conversion webhooks. Does not include any actual changes to the API itself.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #231 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by richardcase** [comment]
Lets update this to v1beta1 (instead of using #230)

**Comment by Danil-Grigorev** [comment]
@richardcase @salasberryfin The PR is updated and the tests are passing. This v1beta1 api has no changes from the current state of v1alpha1, but if we would change or remove some fields, needed conversions could be added later.

**Comment by salasberryfin** [comment]
Thanks @Danil-Grigorev. As discussed earlier today, after this is merged, we'll be adapting the changes in #226 so that new fields in the specification are already in `v1beta``.

**Review by salasberryfin — DISMISSED** [review]

File: controlplane/api/v1alpha1/conversion.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	if err := Convert_v1alpha1_RKE2ControlPlane_To_v1beta1_RKE2ControlPlane(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (dst *RKE2ControlPlane) ConvertFrom(srcRaw conversion.Hub) error {
```
[the remark this answers is not in the document]
Comment (richardcase) [reply controlplane/api/v1alpha1/conversion.go:39]: Also for **RKE2ControlPlaneTemplate**

File: controlplane/api/v1beta1/conversion.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// Hub is a conversion hub for the RKE2ControlPlane resource.
+func (*RKE2ControlPlane) Hub() {}
```
[the remark this answers is not in the document]
Comment (richardcase) [reply controlplane/api/v1beta1/conversion.go:20]: Also for **RKE2ControlPlaneTemplate**

**Comment by Danil-Grigorev** [comment]
@richardcase All types are added and copyright is updated.

**Review by salasberryfin — APPROVED** [review]

**Review by richardcase — APPROVED** [review]

**Pull request merged**